packages/core/src/amazonq/lsp/lspController.ts

```diff
@@ -0,0 +1,378 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs-extra'
+import * as crypto from 'crypto'
+import { getLogger } from '../../shared/logger/logger'
+import { CurrentWsFolders, collectFilesForIndex } from '../../shared/utilities/workspaceUtils'
+import fetch from 'node-fetch'
+import { LspClient } from './lspClient'
+import AdmZip from 'adm-zip'
+import { RelevantTextDocument } from '@amzn/codewhisperer-streaming'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
+import { CodeWhispererSettings } from '../../codewhisperer/util/codewhispererSettings'
+import { activate as activateLsp } from './lspClient'
+import { telemetry } from '../../shared/telemetry'
+import { isCloud9 } from '../../shared/extensionUtilities'
+import { globals } from '../../shared'
+import { AuthUtil } from '../../codewhisperer'
+import { isWeb } from '../../shared/extensionGlobals'
+function getProjectPaths() {
+    const workspaceFolders = vscode.workspace.workspaceFolders
+    if (!workspaceFolders || workspaceFolders.length === 0) {
+        throw Error('No workspace folders found')
+    }
+    return workspaceFolders.map(folder => folder.uri.fsPath)
+}
+
+export interface Chunk {
+    readonly filePath: string
+    readonly content: string
+    readonly context?: string
+    readonly relativePath?: string
+    readonly programmingLanguage?: string
```

Comment:
nit: use `CodewhispererLanguage` instead of vanilla string

Reply:
we should rename it to QLanguage tho

Reply:
The language name and language runtime convertion is a tech debt . If we move these inside LSP then they can be reduced.